### ISSUE TITLE
Example fibonacci.spl won't work

### ISSUE THREAD

**Issue by rickrich** [drsam94/Spl#6]
$ ./spl  ../spl-1.2.1/examples/fibonacci.spl 
[',', 'a', 'lover', 'with', 'an', 'accumulative', 'memory']
Expected a number, but found no noun at line 10

Ref: http://shakespearelang.sourceforge.net/

**Comment by drsam94** [comment]
This issue is similar to a lot of issues I had when first trying to deal with the original implementation of SPL: many of the examples use nouns that are not in the wordlists that they supply!

I added "lover" and "spaceman" as acceptable positive nouns in include/positive_noun.wordlist, and after that I got the example to run properly. So, you should be able to get the same results if you checkout the new version of the wordlists.

(Yes, the error messages probably could be more clear, but the error there is correct: it found no noun, because "lover" was not considered a noun).

**Comment by rickrich** [comment]
Dramatis Personae

This is the section where variables are declared. Each variable can hold a signed integer value and is of the following form:

Name, Description

Where Name is the name of the variable and Description is ignored by the compiler. The compiler will only recognize names that correspond to actual Shakespearean characters.[1]

http://en.wikipedia.org/wiki/Shakespeare_%28programming_language%29#Acts_and_Scenes

You should ignore everything after the comma!

**Comment by drsam94** [comment]
Interesting: that is indeed what is in the spec, though I had implemented it by allowing an initialization there (which is useful, for making it so that you don't have to initialize everyone in early scenes).

I have made a change which is now documented in the README.

Essentially, any declaration that evaluates to a noun phrase will be used as the initial value, and any other declaration will be ignored and the variable is initialized to 0. In theory this is consistent with the initial language spec, as there is undefined behavior if you use a variable before assigning to it, so any resolution (that does not crash, as it did before) is consistent with the spec.
